internals/overlord/checkstate/manager.go

```diff
@@ -387,6 +427,62 @@ type checker interface {
 	check(ctx context.Context) error
 }
 
+func (c *CheckInfo) writeMetrics(writer metrics.Writer) error {
+	checkStatus := 0
+	if c.Status == CheckStatusUp {
+		checkStatus = 1
+	}
+	err := writer.Write(metrics.Metric{
+		Name:       "pebble_check_up",
+		Type:       metrics.TypeGaugeInt,
+		ValueInt64: int64(checkStatus),
+		Comment:    "Whether the health check is up (1) or not (0)",
+		Labels:     []metrics.Label{metrics.NewLabel("check", c.Name)},
+	})
+	if err != nil {
+		return err
+	}
+
+	err = writer.Write(metrics.Metric{
+		Name:       "pebble_perform_check_count",
+		Type:       metrics.TypeCounterInt,
+		ValueInt64: c.PerformCheckCount,
+		Comment:    "Number of times the perform-check has run",
+		Labels:     []metrics.Label{metrics.NewLabel("check", c.Name)},
+	})
+	if err != nil {
+		return err
+	}
+
+	err = writer.Write(metrics.Metric{
+		Name:       "pebble_recover_check_count",
+		Type:       metrics.TypeCounterInt,
+		ValueInt64: c.RecoverCheckCount,
+		Comment:    "Number of times the recover-check has run",
+		Labels:     []metrics.Label{metrics.NewLabel("check", c.Name)},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// WriteMetrics collects and writes metrics for all checks to the provided writer.
+func (m *CheckManager) WriteMetrics(writer metrics.Writer) error {
+	m.checksLock.Lock()
+	defer m.checksLock.Unlock()
+
+	for _, info := range m.checks {
```

Comment:
Per similar comment on `ServiceManager.WriteMetrics`, let's sort by check name to ensure the output is stable.